### DISCUSSION TITLE
Mac trying to clone getting username and password autentication error

### DISCUSSION THREAD

**Discussion by gayasfdev** [community/community#44740] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Username for 'https://github.com': ""
Password for 'https://@github.com': 
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com//New-Batch.git/'

Getting this error while I clone from remote repo to local repo . I havent turned on 2 way autentication . i cleared all the key chain access github but still havning issue .I tried on windows all fine but mac is problem so how can I fix this?

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @gayasfdev ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by airtower-luna** [comment]
> Password for 'https://@github.com':

This line looks like you entered an empty username, which won't work. :wink: 

As for the password, you cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by ScottS2017** [reply]
UPDATE:
Apparently what I did earlier today got associated with my work email address, despite the fact that my personal email address is the primary. 

I hate to say it, but with all of this getting the token attached to the wrong email and that being a blocker, I have to agree with those who want to go back to a password. Tokens are a good idea in theory, but this implementation is too problematic to be live. They need to return to password and work the bugs out of tokens before trying them again. (Not opposed to tokens, just to buggy implementations)

-----------

In my case I do already have the access token and am still getting promted for a username and password. This is confusing since I think it's been years since they swichted to a token, so why is the request for a password still in the system at all?

But my issue is that I can't even perform a fetch, despite the token being properly attached to the user under Preferences > Version Control > Github

Yes, I said a _fetch_. The repo has already been cloned before. So why am I locked out _now_?

I think I remember something from years ago about haveing to put an @ symbol somewhere, in a username? Something like (username)@github(something)

Does that make any sense?

**Reply by airtower-luna** [reply]
> In my case I do already have the access token and am still getting promted for a username and password. This is confusing since I think it's been years since they swichted to a token, so why is the request for a password still in the system at all?

That's because it's generic HTTP authentication in Git, nothing specific to GitHub. Changing the phrasing would be extremely confusing with any Git hoster other than GitHub.

> But my issue is that I can't even perform a fetch, despite the token being properly attached to the user under Preferences > Version Control > Github

This sounds like you're using some GUI instead of using Git directly. In that case you should mention *which* GUI, hopefully someone who knows it will see and be able to help.

> I think I remember something from years ago about haveing to put an @ symbol somewhere, in a username?

You'll occasionally see people recommend adding username and PAT to the URL. It's usually a bad idea to do though, because it makes the PAT show up in plaintext in various places.

> Apparently what I did earlier today got associated with my work email address, despite the fact that my personal email address is the primary.

That has *nothing* to do with the token, only with the author email in the commit(s). The token decides only if you can push, not which account(s) commits will be ascribed to.

**Reply by ScottS2017** [reply]
I'm not using a gui:

<img width="781" alt="Screen Shot 2023-04-03 at 8 55 51 AM" src="https://user-images.githubusercontent.com/32370328/229516635-1fae9557-860f-4087-a4b3-71a78e1a3d77.png">
